sections/staking/components/StakingInfo/MintInfo.tsx

```diff
@@ -28,72 +29,67 @@ const StakingInfo: FC = () => {
-			.plus(stakingAmount)
-			.multipliedBy(targetCRatio)
-			.multipliedBy(SNXRate);
+		const mintAdditionalDebt = stakedCollateral.add(stakingAmount).mul(targetCRatio).mul(SNXRate);
 
-		const changedStakedValue = stakedCollateral.plus(stakingAmount);
+		const changedStakedValue = stakedCollateral.add(stakingAmount);
 
-		const changedTransferable = transferableCollateral.isZero()
-			? zeroBN
+		const changedTransferable = transferableCollateral.eq(0)
+			? wei(0)
 			: getTransferableAmountFromMint(balance, changedStakedValue);
 
 		const changedDebt = mintAdditionalDebt;
 
-		const changedSUSDBalance = sUSDBalance.plus(amountToMintBN);
+		const changedSUSDBalance = sUSDBalance.add(amountToMintBN);
 
-		const changeCRatio = currentCRatio.isLessThan(targetCRatio)
-			? unstakedCollateral
-					.plus(stakedCollateral)
-					.multipliedBy(SNXRate)
-					.dividedBy(mintAdditionalDebt)
-					.multipliedBy(100)
-			: changedStakedValue.multipliedBy(SNXRate).dividedBy(mintAdditionalDebt).multipliedBy(100);
+		const changeCRatio = mintAdditionalDebt.gt(0)
+			? currentCRatio.lt(targetCRatio)
+				? unstakedCollateral.add(stakedCollateral).mul(SNXRate).div(mintAdditionalDebt).mul(100)
+				: changedStakedValue.mul(SNXRate).div(mintAdditionalDebt).mul(100)
+			: wei(0);
 
 		return {
 			barRows: [
 				{
 					title: t('staking.info.table.staked'),
 					value: sanitiseValue(stakedCollateral),
 					changedValue: sanitiseValue(changedStakedValue),
-					percentage: collateral.isZero()
-						? toBigNumber(0)
-						: sanitiseValue(stakedCollateral).dividedBy(collateral),
-					changedPercentage: collateral.isZero()
-						? toBigNumber(0)
-						: sanitiseValue(changedStakedValue).dividedBy(collateral),
+					percentage: collateral.eq(0) ? wei(0) : sanitiseValue(stakedCollateral).div(collateral),
+					changedPercentage: collateral.eq(0)
+						? wei(0)
+						: sanitiseValue(changedStakedValue).div(collateral),
 					currencyKey: CryptoCurrency.SNX,
 				},
 				{
 					title: t('staking.info.table.transferable'),
 					value: sanitiseValue(transferableCollateral),
 					changedValue: sanitiseValue(changedTransferable),
-					percentage: collateral.isZero()
-						? toBigNumber(0)
-						: sanitiseValue(transferableCollateral).dividedBy(sanitiseValue(collateral)),
-					changedPercentage: collateral.isZero()
-						? toBigNumber(0)
-						: sanitiseValue(changedTransferable).dividedBy(sanitiseValue(collateral)),
+					percentage: collateral.eq(0)
+						? wei(0)
+						: sanitiseValue(transferableCollateral).div(sanitiseValue(collateral)),
+					changedPercentage: collateral.eq(0)
+						? wei(0)
+						: sanitiseValue(changedTransferable).div(sanitiseValue(collateral)),
 					currencyKey: CryptoCurrency.SNX,
 				},
 			],
 			dataRows: [
 				{
 					title: t('staking.info.table.c-ratio'),
-					value: sanitiseValue(toBigNumber(100).dividedBy(currentCRatio)),
+					value: currentCRatio.eq(0) ? wei(0) : sanitiseValue(wei(100).div(currentCRatio)),
```

Comment:
same question here with sanitiseValue

Reply:
we have to do this or else code fails with divide by zero

ideally we wait until all queries are loaded before even trying to go to this section but one step at a time, right?